### PR TITLE
tools/op: add switch command

### PR DESCRIPTION
**Pull request by adeebshihadeh**
Copied from my `nuke` [helper](https://github.com/adeebshihadeh/config/blob/29a667520fd7ccc2bff6a63e5a3ecae98817692a/.bash_profile#L25) I use for development.

This makes it easier to switch between release branches (with no submodules) and development branches:
```
op swtich release3
op switch master
```

### CODE REVIEW

**Comment by adeebshihadeh** [comment]
@maxime-desroches merge if this looks good to you

**Comment by maxime-desroches** [comment]
Trying `op switch nightly`:
```
batman:/tmp/openpilot$ op switch nightly
Checking system → [✔]
remote: Enumerating objects: 1367, done.
remote: Counting objects: 100% (1367/1367), done.
remote: Compressing objects: 100% (1010/1010), done.
remote: Total 1187 (delta 201), reused 938 (delta 145), pack-reused 0 (from 0)
Receiving objects: 100% (1187/1187), 141.94 MiB | 19.83 MiB/s, done.
Resolving deltas: 100% (201/201), completed with 39 local objects.
From github.com:commaai/openpilot
 * branch            nightly    -> FETCH_HEAD
error: pathspec 'nightly' did not match any file(s) known to git
```

File: tools/op.sh [hunk 1]
```diff
@@ -312,6 +312,23 @@ function op_sim() {
...
 
+function op_switch() {
+  op_before_cmd
+
+  if [ -z "$1" ]; then
+    echo "specify a branch"
+    exit 1
+  fi
+
+  git fetch origin $1
```
Comment (maxime-desroches) [inline tools/op.sh:323]:
```suggestion
  git fetch origin $1:$1
```

File: tools/op.sh [hunk 2]
```diff
@@ -312,6 +312,23 @@ function op_sim() {
...
+  git checkout -f $1
+  git reset --hard origin/$1
```
Comment (maxime-desroches) [inline tools/op.sh:325]:
```suggestion
  git reset --hard $1
```

**Comment by maxime-desroches** [comment]
This will also fails when switching from `release3` to `master` since we use `selfdrive/modeld/models/supercombo.onnx` to verify if the LFS files are present or not.

We can switch to checking `selfdrive/modeld/models/dmonitoring_model.onnx` instead

**Comment by adeebshihadeh** [comment]
> Trying `op switch nightly`:
> 
> ```
> batman:/tmp/openpilot$ op switch nightly
> Checking system → [✔]
> remote: Enumerating objects: 1367, done.
> remote: Counting objects: 100% (1367/1367), done.
> remote: Compressing objects: 100% (1010/1010), done.
> remote: Total 1187 (delta 201), reused 938 (delta 145), pack-reused 0 (from 0)
> Receiving objects: 100% (1187/1187), 141.94 MiB | 19.83 MiB/s, done.
> Resolving deltas: 100% (201/201), completed with 39 local objects.
> From github.com:commaai/openpilot
>  * branch            nightly    -> FETCH_HEAD
> error: pathspec 'nightly' did not match any file(s) known to git
> ```

Ah, likely due to the git config? Did you clone with `-b nightly`?

**Comment by adeebshihadeh** [comment]
> This will also fails when switching from `release3` to `master` since we use `selfdrive/modeld/models/supercombo.onnx` to verify if the LFS files are present or not.
> 
> We can switch to checking `selfdrive/modeld/models/dmonitoring_model.onnx` instead

There's no LFS on release, so we shouldn't have any LFS files.

**Comment by morrislee** [comment]
This is what I use to completely switch from one branch/fork including the agnos helped by yassine

`cd /data; rm -rf ./openpilot; git clone -b sl_model --depth 1 --single-branch --recurse-submodules --shallow-submodules https://github.com/commaai/openpilot.git openpilot; git reset --hard; git pull --recurse-submodules; cd openpilot; pkill -f thermald; ./system/hardware/tici/agnos.py ./system/hardware/tici/agnos.json --swap; rm -f prebuilt;git reset --hard; git pull --recurse-submodules; sudo reboot`

**Comment by sshane** [comment]
btw I use `upstream` for commaai/openpilot, `origin` for my fork

**Pull request merged**